### PR TITLE
[native] Add session properties for scale writer query configs

### PR DESCRIPTION
**Pull request by duxiao1212**
## Motivation and Context
Add session properties for scale writer query configs

```
scaled_writer_rebalance_max_memory_usage_ratio
scaled_writer_max_partitions_per_writer
scaled_writer_min_partition_processed_bytes_rebalance_threshold
scaled_writer_min_processed_bytes_rebalance_threshold
```

## Impact

Low impact

## Contributor checklist

- [x] Please make sure your submission complies with our [contributing guide](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md), in particular [code style](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#code-style) and [commit standards](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#commit-standards).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/SessionProperties.h [hunk 1]
```diff
@@ -178,6 +178,31 @@ class SessionProperties {
   static constexpr const char* kSelectiveNimbleReaderEnabled =
       "native_selective_nimble_reader_enabled";
 
+  /// The max ratio of a query used memory to its max capacity, and the scale
+  /// writer exchange stops scaling writer processing if the query's current
+  /// memory usage exceeds this ratio. The value is in the range of (0, 1].
+  static constexpr const char* kScaleWriterRebalanceMaxMemoryUsageRatio =
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/SessionProperties.h:184]: You need native prefix like native_selective_nimble_reader_enabled

**Comment by kewang1024** [comment]
We also need to add them to coordinator
An example: https://github.com/prestodb/presto/pull/23741

Can we squash the commits to one?

**Comment by pdabre12** [comment]
On the coordinator side, the native session properties have been moved to `NativeWorkerSessionPropertyProvider`, an example: https://github.com/prestodb/presto/pull/23891

**Comment by duxiao1212** [comment]
> On the coordinator side, the native session properties have been moved to `NativeWorkerSessionPropertyProvider`, an example: #23891

Thanks, @pdabre12 . However, I noticed that NativeWorkerSessionPropertyProvider is marked as deprecated, and there doesn't seem to be any replacement suggestion. This is causing some confusion. Should we remove the deprecated annotation or do you have any recommendations?"

**Comment by pdabre12** [comment]
> > On the coordinator side, the native session properties have been moved to `NativeWorkerSessionPropertyProvider`, an example: #23891
> 
> Thanks, @pdabre12 . However, I noticed that NativeWorkerSessionPropertyProvider is marked as deprecated, and there doesn't seem to be any replacement suggestion. This is causing some confusion. Should we remove the deprecated annotation or do you have any recommendations?

It is marked as deprecated, because we support the retrieval of the session properties through the sidecar, the PR: :https://github.com/prestodb/presto/pull/23045. Ideally, we want to use this functionality but this is a temporary class where native session properties were re-located to until the adoption of the sidecar.

**Review by zation99 — DISMISSED** [review]
lgtm

**Review by xiaoxmeng — APPROVED** [review]
@duxiao1212 thanks for the change!

**Review by arhimondr — APPROVED** [review]

**Pull request closed without merging**